### PR TITLE
#165305731 Opt in/out of notifications

### PR DESCRIPTION
**Pull request by a-braham**
#### What does this PR do?
- Creates a UI/UX for users to opt in/out of notifications through in-app and email notifications
#### Description of Task to be completed?
- User can choose the type of notifications to get either push or email
#### How should this be manually tested?

- After cloning the repo and installing dependencies `npm install`, run `ft-user-notifications-165305731`
- Then run `npm run start:dev`
- click settings

#### What are the relevant pivotal tracker stories?
[#165305731](https://www.pivotaltracker.com/n/projects/2326460/stories/165305731)
#### Screenshots (if appropriate)
Screenshot 2019-06-25 at 23 38 30

### CODE REVIEW

**Review by matthenge — APPROVED** [review]
LGTM

**Review by Amoswachira — APPROVED** [review]

File: src/components/notifications/NotificationComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal } from 'react-bootstrap';
+import '../../styles/style.css';
+import push from '../../services/images/push.png';
+import email from '../../services/images/email.png';
+
+const NotificationModal = ({ show, closeModal, onToggle, notification }) => {
```
Comment (charisschomba) [inline src/components/notifications/NotificationComponent.jsx]: If you are destructuring more than 3 props you should have them in a new line

File: src/components/notifications/NotificationComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+              <img src={push} alt="push" className="notif-icons" />
+              <h6 className="notif-title">
+                <small>
+                  <b>
+                    Push
+                  </b>
+                </small>
+              </h6>
+              {Object.values(notification)[0].in_app_notifications ? (
+                <>
```
Comment (charisschomba) [inline src/components/notifications/NotificationComponent.jsx:41]: why do you have this empty tags?

File: src/components/notifications/NotificationComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                  <h6 className="notif-status-app">
+                    <small>
+                      <b>
+                        On
+                      </b>
+                    </small>
+                  </h6>
+                  <label className="switch">
+                    <input type="checkbox" className="checks" name="in_app_notifications" checked onChange={(event) => onToggle(event, false)} />
```
Comment (charisschomba) [inline src/components/notifications/NotificationComponent.jsx]: Have these input properties in a new line, each for easier readability.

File: src/redux/actions/notifications.action.js [hunk 4]
```diff
@@ -61,3 +61,45 @@ export const clearNotify = () => async (dispatch) => {
...
+      headers: {
+        'content-type': 'application/json',
+        Authorization: `Bearer ${token}`
+      },
+    };
+    const response = await fetchNotifStatus(axiosHeader);
+    dispatch(optInOutSuccess(response.data));
+  } catch(error) {
+    dispatch(optInOutError('Notififications cannot be found'));
+  }
```
Comment (charisschomba) [inline src/redux/actions/notifications.action.js:89]: You should also handle network errors, eg if there is no internet connection, the app should not crush.

File: src/redux/actions/notifications.action.js [hunk 5]
```diff
@@ -61,3 +61,45 @@ export const clearNotify = () => async (dispatch) => {
...
+    const axiosHeader = {
+      headers: {
+        'content-type': 'application/json',
+        Authorization: `Bearer ${token}`
+      },
+    };
+    const response = await optInOutApi(data, axiosHeader);
+    dispatch(optInOutSuccess(response.data));
+  } catch(error) {
+    dispatch(optInOutError('Cannot senf notififications subscription'));
```
Comment (charisschomba) [inline src/redux/actions/notifications.action.js]: same here and there is a typo `senf`

**Review by charisschomba — CHANGES_REQUESTED** [review]

**Review by Ogutu-Brian — APPROVED** [review]

**Pull request merged**